atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/priority/SweepPriorityCalculator.java

```diff
@@ -107,6 +117,53 @@ public SweepPriorityCalculator(KeyValueService kvs, SweepPriorityStore sweepPrio
         return scores;
     }
 
+    private boolean decideWhetherToLogAllPriorities() {
+        ZonedDateTime now = ZonedDateTime.now();
+        boolean shouldLog = Duration.between(lastLogTime, now).compareTo(Duration.of(10, ChronoUnit.MINUTES)) > 0;
+        if (shouldLog) {
+            lastLogTime = now;
+        }
+
+        return shouldLog;
+    }
+
+    private void logSweepPriority(boolean shouldLog, TableReference tableRef,
+            SweepPriority oldPriority, SweepPriority newPriority, double priorityScore) {
+        if (shouldLog) {
+            log.debug("Calculated sweep priority score "
+                            + "[tableRef={},"
+                            + "priorityScore={},"
+
+                            + "oldPriority_lastSweepTimeMillis={},"
+                            + "oldPriority_minimumSweptTimestamp={},"
+                            + "oldPriority_cellTsPairsExamined={},"
+                            + "oldPriority_staleValuesDeleted={},"
+                            + "oldPriority_writeCount={},"
+
+                            + "newPriority_lastSweepTimeMillis={},"
+                            + "newPriority_minimumSweptTimestamp={},"
+                            + "newPriority_cellTsPairsExamined={},"
+                            + "newPriority_staleValuesDeleted={},"
+                            + "newPriority_writeCount={}]",
+
+                    LoggingArgs.tableRef(tableRef),
+                    SafeArg.of("priorityScore", priorityScore),
+
+                    SafeArg.of("oldPriority_lastSweepTimeMillis", oldPriority.lastSweepTimeMillis()),
+                    SafeArg.of("oldPriority_minimumSweptTimestamp", oldPriority.minimumSweptTimestamp()),
+                    SafeArg.of("oldPriority_cellTsPairsExamined", oldPriority.cellTsPairsExamined()),
+                    SafeArg.of("oldPriority_staleValuesDeleted", oldPriority.staleValuesDeleted()),
+                    SafeArg.of("oldPriority_writeCount", oldPriority.writeCount()),
+
+                    SafeArg.of("newPriority_lastSweepTimeMillis", newPriority.lastSweepTimeMillis()),
```

Comment:
nit: we can refactor this to generate logging for old and new priorities as this a bunch of duplicate code. Something like `getLogMesasageAndArgsForPriority("old"/"new", priority)` If we add a new parameter for priority calculation, we'll have to update both places in the future.

Reply:
This is kind of messy because we have to generate the 2 fragments, concatenate the  strings and merge the arrays.  Given we've only duplicated this code once I don't think it is worth the effort and reduced readability to refactor this.  If we end up logging many historical versions then we can think more about how to do this.